### PR TITLE
Release Hermes v1.2.0

### PR DESCRIPTION
**Pull request by romac**
Closes: #2941

### CODE REVIEW

**Review by seanchen1991 — APPROVED** [review]
I addressed the `cargo doc` issues that came up. Otherwise, it looks great to me. Thanks @romac!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,102 @@
...
+Additionally, IBC clients with a trust level between `1/3` and `2/3` inclusive are now allowed.
+
+### General
+
+#### BREAKING CHANGES
+
+- Update minimum supported Rust version to 1.65.0
+  ([#2817](https://github.com/informalsystems/hermes/pull/2817))
+- Update to `tendermint-rs` 0.27 and ibc-proto 0.23
+  ([#2855](https://github.com/informalsystems/ibc-rs/issues/2859))
```
Comment (ljoss17) [inline CHANGELOG.md:21]:
```suggestion
  ([#2855](https://github.com/informalsystems/hermes/issues/2859))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,102 @@
...
+  ([#2893](https://github.com/informalsystems/ibc-rs/pull/2893))
```
Comment (ljoss17) [inline CHANGELOG.md:22]:
```suggestion
  ([#2893](https://github.com/informalsystems/hermes/pull/2893))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,102 @@
...
+
+#### IMPROVEMENTS
+
+- Test against Gaia v8 in CI
+  ([#2820](https://github.com/informalsystems/hermes/issues/2820))
+- Use rolling Ubuntu as base image in Hermes Docker image to
+  reduce fix several vulnerabilities found in previous base image
+  ([#2810](https://github.com/informalsystems/ibc-rs/issues/2810))
```
Comment (ljoss17) [inline CHANGELOG.md:30]:
```suggestion
  ([#2810](https://github.com/informalsystems/hermes/issues/2810))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,102 @@
...
+
+### Hermes - `ibc-relayer-cli` (v1.2.0)
+
+#### IMPROVEMENTS
+
+- Collate consecutive heights and sequence numbers shown in logs
+  ([#2847](https://github.com/informalsystems/ibc-rs/issues/2847))
```
Comment (ljoss17) [inline CHANGELOG.md:37]:
```suggestion
  ([#2847](https://github.com/informalsystems/hermes/pull/2847))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,102 @@
...
+
+### Relayer Library - `ibc-relayer` (v0.21.0)
+
+### FEATURES
+
+- Add support for Ed25519 keys
+  ([#2851](https://github.com/informalsystems/ibc-rs/issues/2851))
```
Comment (ljoss17) [inline CHANGELOG.md:44]:
```suggestion
  ([#2851](https://github.com/informalsystems/hermes/issues/2851))
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,102 @@
...
+
+### BUG FIXES
+
+- Ensure Hermes uses Rustls instead of OpenSSL for TLS on all platforms
+  ([#2799](https://github.com/informalsystems/ibc-rs/issues/2799))
```
Comment (ljoss17) [inline CHANGELOG.md:49]:
```suggestion
  ([#2799](https://github.com/informalsystems/hermes/issues/2799))
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,102 @@
...
+- Fix an issue where Hermes would sometimes fails to retrieve the
+  begin/end block events because of a quirk of the Tendermint event indexer
+  ([#2867](https://github.com/informalsystems/ibc-rs/issues/2867))
```
Comment (ljoss17) [inline CHANGELOG.md:52]:
```suggestion
  ([#2867](https://github.com/informalsystems/hermes/issues/2867))
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,102 @@
...
+  setting against the full node's `min_gas_price` setting, ensuring that
+  the former is at least equal to the latter
+  ([#2776](https://github.com/informalsystems/hermes/issues/2776))
+- IBC clients with trust thresholds in the range [1/3, 2/3] (inclusive) are now allowed
+  ([#2798](https://github.com/informalsystems/hermes/issues/2798))
+- Move default implementations of `init_event_monitor`, `id`, `get_key`,
+  and `add_key` from CosmosSdkChain to ChainEndpoint, and change
+  `ChainEndpoint::config()` to return a `&ChainConfig` instead of
+  a `ChainConfig`
+  ([#2806](https://github.com/informalsystems/ibc-rs/issues/2806))
```
Comment (ljoss17) [inline CHANGELOG.md:69]:
```suggestion
  ([#2806](https://github.com/informalsystems/hermes/issues/2806))
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,102 @@
...
+- Remove `init_event_monitor` from `ChainEndpoint` in favor of `subscribe` method
+  ([#1456](https://github.com/informalsystems/ibc-rs/issues/1456))
```
Comment (ljoss17) [inline CHANGELOG.md:71]:
```suggestion
  ([#1456](https://github.com/informalsystems/hermes/issues/1456))
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,102 @@
...
+- Added the possibility to specify multiple chains for integration tests
+  ([#2003](https://github.com/informalsystems/ibc-rs/issues/2003))
```
Comment (ljoss17) [inline CHANGELOG.md:73]:
```suggestion
  ([#2003](https://github.com/informalsystems/hermes/issues/2003))
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,102 @@
...
+  - Move `from_seed_file` into `KeyEntry`
+  - Simplify the `get_address` implementation
+
+
+### Integration Test Framework - [`ibc-test-framework`](tools/test-framework) (v0.21.0)
+
+#### FEATURES
+
+- Added integration tests and guide entry for packet forwarding.
+  ([#1983](https://github.com/informalsystems/ibc-rs/issues/1983))
```
Comment (ljoss17) [inline CHANGELOG.md:86]:
```suggestion
  ([#1983](https://github.com/informalsystems/hermes/issues/1983))
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,102 @@
...
+- Improved error logs when the height given to upgrade a client is too low.
+  ([#2781](https://github.com/informalsystems/hermes/issues/2781))
+- Added integration tests for client upgrade.
+  ([#2312](https://github.com/informalsystems/ibc-rs/issues/2312))
```
Comment (ljoss17) [inline CHANGELOG.md:90]:
```suggestion
  ([#2312](https://github.com/informalsystems/hermes/issues/2312))
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,102 @@
...
+
+
+### Telemetry & Metrics - [`ibc-telemetry`](crates/telemetry) (v0.21.0)
+
+#### IMPROVEMENTS
+
+- Add support for dumping Prometheus metrics to JSON
+  ([#2890](https://github.com/informalsystems/ibc-rs/issues/2890))
```
Comment (ljoss17) [inline CHANGELOG.md:98]:
```suggestion
  ([#2890](https://github.com/informalsystems/hermes/issues/2890))
```

**Review by ljoss17 — APPROVED** [review]
Seems good to me too, just some nits with the issues/PRs links in the changelog

**Pull request merged**